### DISCUSSION TITLE
How to use workflows to edit files

### DISCUSSION THREAD

**Discussion by Matronicz** [community/community#52871] [Actions] [answered]
So, in my workflow, a python file runs, then saves data to a file using open(). the file with the same name is available in my repository. How can I make my original file change to the file generated by python?

**Comment by airtower-luna** [comment, marked as answer]
The same way you'd do it locally: After modifying the file, add, commit, and push it. Before that set the user name and email appropriately, usually so the commit is authored by the Actions bot. The [`actions/checkout` documentation has an example](https://github.com/actions/checkout#push-a-commit-using-the-built-in-token). I'd advise against using `git add .` (as the example does) though, instead add the file(s) you want to change only.
